process-document/src/main/kotlin/com/ritense/processdocument/service/CaseTaskListSearchService.kt

```diff
@@ -0,0 +1,241 @@
+package com.ritense.processdocument.service
+
+import com.ritense.case.domain.TaskListColumn
+import com.ritense.case.repository.TaskListColumnRepository
+import com.ritense.document.domain.impl.JsonDocumentContent
+import com.ritense.document.domain.impl.JsonSchemaDocument
+import com.ritense.document.domain.impl.JsonSchemaDocumentDefinitionId
+import com.ritense.document.domain.impl.JsonSchemaDocumentId
+import com.ritense.processdocument.domain.CaseTask
+import com.ritense.processdocument.web.result.TaskListRowDto
+import com.ritense.valtimo.camunda.domain.CamundaExecution
+import com.ritense.valtimo.camunda.domain.CamundaTask
+import com.ritense.valtimo.camunda.repository.CamundaTaskSpecificationHelper
+import com.ritense.valtimo.contract.authentication.UserManagementService
+import com.ritense.valtimo.contract.database.QueryDialectHelper
+import com.ritense.valtimo.service.CamundaTaskService
+import com.ritense.valtimo.service.CamundaTaskService.TaskFilter
+import com.ritense.valueresolver.ValueResolverService
+import jakarta.persistence.EntityManager
+import jakarta.persistence.criteria.CriteriaBuilder
+import jakarta.persistence.criteria.CriteriaQuery
+import jakarta.persistence.criteria.Expression
+import jakarta.persistence.criteria.JoinType
+import jakarta.persistence.criteria.Order
+import jakarta.persistence.criteria.Path
+import jakarta.persistence.criteria.Predicate
+import jakarta.persistence.criteria.Root
+import java.util.UUID
+import java.util.stream.Collectors
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Pageable
+import org.springframework.data.domain.Sort
+
+
+class CaseTaskListSearchService(
+    private val entityManager: EntityManager,
+    private val valueResolverService: ValueResolverService,
+    private val taskListColumnRepository: TaskListColumnRepository,
+    private val userManagementService: UserManagementService,
+    private val queryDialectHelper: QueryDialectHelper
+) {
+    private val CONTENT = "content"
+    private val INTERNAL_STATUS = "internalStatus"
+    private val INTERNAL_STATUS_ORDER = "internalStatus.order"
+    private val DOC_PREFIX = "doc:"
+    private val CASE_PREFIX = "case:"
+    private val DOCUMENT_FIELD_MAP: Map<String, String> = mapOf(
+        "definitionId.name" to "documentDefinitionId.name",
+        "definitionId.version" to "documentDefinitionId.key",
+        INTERNAL_STATUS to INTERNAL_STATUS_ORDER
+    )
+
+    fun getTasksByCaseDefinition(caseDefinitionName: String, assignmentFilter: CamundaTaskService.TaskFilter, pageable: Pageable): Page<TaskListRowDto> {
+        // No authorization on this level, as we have to fully rely on the documentSearchService for filtering results
```

Comment:
documentSearchService?

Im missing logic to filter the task list. Users should only see tasks that they have permission to. This is also done for the existing task list. see `CamundaTaskService.findTasksFiltered(..)`

Reply:
Added the PBAC filtering.